### PR TITLE
Manifests

### PR DESCRIPTION
**Pull request by ianmkenney**
Implement the use of Manifest files to build states in testing.

### CODE REVIEW

**Comment by ianmkenney** [comment]
@orbeckst I have included very basic tests that increase coverage quite a bit while using manifest_parser to define and build states. I will build more tests in different branches; I have overstayed my welcome in this one, as it was only really intended for implementation of manifests. Merge if everything looks good to you!

File: mdpow/tests/test_equilibration_script.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+from mdpow.run import equilibrium_simulation
+from mdpow.config import get_configuration
+
+class TestEquilibriumScript(object):
+    
+    def setup(self):
+        self.tmpdir = td.TempDir()
+        self.old_path = os.getcwd()
+        self.resources = self.old_path + "/mdpow/tests/testing_resources"
```
Comment (orbeckst) [inline mdpow/tests/test_equilibration_script.py]: use `os.path.join()`

File: mdpow/tests/test_equilibration_script.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        m = manifest.Manifest(os.path.join(self.resources,'manifest.yml'))
+        m.assemble('base',self.tmpdir.name)
+
+    def _run_equil(self, solvent, dirname):
+        try:
+            print(os.listdir('.'))
```
Comment (orbeckst) [inline mdpow/tests/test_equilibration_script.py]: no `print` – use logging if you must.

File: mdpow/tests/test_equilibration_script.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            cfg = get_configuration('runinput.yml')
+            self.S = equilibrium_simulation(cfg, solvent, dirname=dirname)
+        except:
+            assert False
```
Comment (orbeckst) [inline mdpow/tests/test_equilibration_script.py]: just
```python
raise AssertionError
```

Comment (orbeckst) [inline mdpow/tests/test_equilibration_script.py]: This tests if the simulation can be run. Ok for a start but we eventually want to test for regression and correctness. Add a comment.

Also let the assertion say something like
```python
raise AssertionError("equilibrium_simulation() failed to run")
```

File: mdpow/tests/test_equilibration_script.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    def test_basic_run(self):
+        os.chdir(self.tmpdir.name)
+        try:
+            self._run_equil('water','benzene/')
+            self._new_structures()
+        except:
+            assert False
```
Comment (orbeckst) [inline mdpow/tests/test_equilibration_script.py]: raise AssertionError

File: mdpow/tests/test_equilibration_script.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    def _run_equil(self, solvent, dirname):
+        try:
+            print(os.listdir('.'))
+            cfg = get_configuration('runinput.yml')
+            self.S = equilibrium_simulation(cfg, solvent, dirname=dirname)
+        except:
+            assert False
+
+    def test_basic_run(self):
+        os.chdir(self.tmpdir.name)
```
Comment (orbeckst) [inline mdpow/tests/test_equilibration_script.py]: Possibly use a `with` contextmanager from `tmpdir`?

Comment (orbeckst) [inline mdpow/tests/test_equilibration_script.py]: Or we should at least create our own context manager, eg [gromacs.utilties.in_dir](https://github.com/Becksteinlab/GromacsWrapper/blob/develop/gromacs/utilities.py#L256):
```python
with in_dir(self.tmpdir.name, create=False):
   do_stuff
```
and in won't destroy the directory.

Either import from GW or copy the definition.

File: mdpow/tests/test_fep.py [hunk 6]
```diff
@@ -3,6 +3,7 @@
 
 import mdpow
 import mdpow.fep
+import manifest
```
Comment (orbeckst) [inline mdpow/tests/test_fep.py]: needed?

File: mdpow/tests/test_fep_script.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import manifest
+from mdpow.equil import Simulation
+
+from mdpow.run import fep_simulation
+from mdpow.config import get_configuration
+
+class TestFEPScript(object):
+    
+    def setup(self):
+        self.tmpdir = td.TempDir()
```
Comment (orbeckst) [inline mdpow/tests/test_fep_script.py]: Test directories should autodestruct after tests are done. Are they? Check `tmpdir`, it can do this kind of stuff.

Comment (orbeckst) [inline mdpow/tests/test_fep_script.py]: As we just checked: it *should* work with `tempdir.TempDir` when it's gc'ed. (If not then that's a bug in `tempdir`...)

**Comment by orbeckst** [comment]
See comments inline – just small things.

However, do we need the binary XTC and TRR files? What sizes are they?

**Comment by orbeckst** [comment]
Goal of this PR: get test framework in place. We will later modify tests to test for correctness.

**Comment by ianmkenney** [comment]
@orbeckst I think I got everything we talked about

**Comment by orbeckst** [comment]
Can you squash some commits so that the big files removed in f75854e905c6ccdefed35bc39898e3f2dad00918 never show up in the history?

Also see if you can squash other commits together. (Don't squash *everything*, each commit should be fairly self contained. Squash 316dc9b and  bcf36f4 together (nuking fep_opls.mdp could have been on its own PR);  like  		bbbc89b could be  squashed with previous ones). Have fun with `git rebase -i`....

Add a comment in the PR along the lines of "manually rebased and force-pushed".

**Comment by orbeckst** [comment]
Apart from the history rewriting, this looks very good!

**Comment by orbeckst** [comment]
@ianmkenney let me know when you have rewritten the history and are ready to merge.

**Comment by ianmkenney** [comment]
@orbeckst it has been rewritten. take a look

**Comment by orbeckst** [comment]
Good work @ianmkenney and nice history rewriting. Merging now.

**Pull request merged**